sdk/eventhub/Azure.Messaging.EventHubs/MIGRATIONGUIDE.md

```diff
@@ -0,0 +1,380 @@
+| `EventHubClient.SendAsync()`                          | `EventHubProducerClient.SendAsync()`                               | [Publish events](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample07_PublishEventsWithCustomMetadata.cs) |
+
+### Read events 
+
+| In v4                                          | Equivalent in v5                                                 | Sample |
+|------------------------------------------------|------------------------------------------------------------------|--------|
+| `PartitionReceiver.ReceiveAsync()`                       | `EventHubConsumerClient.ReadEventsFromPartitionAsync()`                               | [Read events](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ReadEventsFromAKnownPosition.cs) |
+| `EventHubClient.SetReceiveHandler()`                       | `EventHubConsumerClient.ReadEventsFromPartitionAsync()`                               | [Read events](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample09_ReadEventsFromAKnownPosition.cs) |
+
+### Minor renames
+
+| In v4                                          | Equivalent in v5            |
+|------------------------------------------------|-----------------------------|
+| `EventPosition.FromStart()`                    | `EventPosition.Earliest`  |
+| `EventPosition.FromEnd()`                      | `EventPosition.Latest`    |
+
+## Migration samples
+
+ - [Migrating code from `PartitionSender` to `EventHubProducerClient` for publishing events to a partition](#migrating-code-from-partitionsender-to-eventhubproducerclient-for-publishing-events-to-a-partition)
+ - [Migrating code from `EventHubClient` to `EventHubProducerClient` for publishing events using automatic routing](#migrating-code-from-eventhubclient-to-eventhubproducerclient-for-publishing-events-using-automatic-routing)
+ - [Migrating code from `EventHubClient` to `EventHubProducerClient` for publishing events with partition key](#migrating-code-from-eventhubclient-to-eventhubproducerclient-for-publishing-events-with-partition-key)
+ - [Migrating code from `PartitionReceiver` to `EventHubConsumerClient` for reading events in batches](#migrating-code-from-partitionreceiver-to-eventhubconsumerclient-for-reading-events-in-batches)
+ - [Migrating code from `EventProcessorHost` to `EventProcessorClient` for receiving events](#migrating-code-from-eventprocessorhost-to-eventprocessorclient-for-reading-events)
+
+### Migrating code from `PartitionSender` to `EventHubProducerClient` for publishing events to a partition
+
+In v4, events could be published to a single partition using `PartitionSender`.
```

Comment:
I think this is a great place to detail that they could also send a single event, a set of events, or an explicit batch.    We are only supporting batch currently to ensure that there are no unexpected exceptions generated during send;  you can't put it in a batch if it was too large to send.

Reply:
Agreed!